logstash-core/lib/logstash/api/lib/app/commands/system/plugins_command.rb

```diff
@@ -0,0 +1,26 @@
+# encoding: utf-8
+require "app/command"
+
+class LogStash::Api::PluginsCommand < LogStash::Api::Command
+
+  def run
+    { :total => plugins.count, :plugins => plugins }
+  end
+
+  private
+
+  def plugins
+    @plugins ||= find_plugins_gem_specs.map do |spec|
+      { :name => spec.name, :version => spec.version.to_s }
+    end
+  end
+
+  def find_plugins_gem_specs
+    @specs ||= Gem::Specification.find_all.select{|spec| logstash_plugin_gem_spec?(spec)}
```

Comment:
Is a good idea, but in this case I think the code is clear enough, so not much of need to make it shorter ... I'm a fan of clear and easy to understand code :+1: